### PR TITLE
Feature/create messagehistoryandmessage controller

### LINKED ISSUE #45: create message history and message controller

### PR DESCRIPTION
**Pull request by IruomaJennifer**

### CODE REVIEW

File: CCSA_ChatApp.Db/Repositories/MessageRepository.cs [hunk 1]
```diff
@@ -39,37 +39,39 @@ private async new Task  Commit()
...
+            var sender = await _userRepository.GetUserById(senderId);
             var reciever = await _userRepository.GetUserByUsername(receiverUsername);
             if (sender != null)
             {
                 if (reciever != null)
                 {
                     var messageHistory = new MessageHistory { Receiver = reciever, Sender = sender };
-                    _session.Save(message);
-                    _messageHistoryRepository.CreateMessageHistory(message, messageHistory);
+                    await _session.SaveAsync(message);
```
Comment (Adexandria) [inline CCSA_ChatApp.Db/Repositories/MessageRepository.cs]: You can join line 46 and 48 together to remove the nested if

File: CCSA_ChatApp.Db/Repositories/MessageRepository.cs [hunk 2]
```diff
@@ -39,37 +39,39 @@ private async new Task  Commit()
...
             var groupChat = await _groupChatRepository.GetGroupChatById(groupId);
             
             if (sender != null)
             {
                 if (groupChat != null)
                 {
                     var messageHistory = new MessageHistory { GroupChatUser = groupChat, Sender = sender };
-                    _session.Save(message);
-                    _messageHistoryRepository.CreateMessageHistory(message, messageHistory);
+                    await _session.SaveAsync(message);
```
Comment (Adexandria) [inline CCSA_ChatApp.Db/Repositories/MessageRepository.cs:65]: Same here line 64 and 66

File: CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+
+namespace CCSA_ChatApplication.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class MessageAndHistoryController : ControllerBase
```
Comment (Adexandria) [inline CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs]: Change your controller name 
Maybe to message history

File: CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        private readonly MessageService _messageService;
+        private readonly MessageHistoryService _messageHistoryService;
+        public MessageAndHistoryController(MessageService messageService, MessageHistoryService messageHistoryService)
+        {
+            _messageService = messageService;
+            _messageHistoryService = messageHistoryService;
+        }
+
+        [HttpDelete("message")]
```
Comment (Adexandria) [inline CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs]: Pls pass your message id as a path

File: CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        public async Task<IActionResult> DeleteMessageByMessageId(Guid messageId)
+        {
+           await _messageService.DeleteMessageByMessageId(messageId);
+           return Ok("Message has been deleted successfully");
+        }
+
+        [HttpGet("messages/group")]
+        public IActionResult FetchGroupChatMessagesByGroupId(Guid groupId)
+        {
```
Comment (Adexandria) [inline CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs]: Same here pass your group Id as a path

File: CCSA_ChatApp.Db/Repositories/MessageRepository.cs (hunk 1 above)
Comment (IruomaJennifer) [reply CCSA_ChatApp.Db/Repositories/MessageRepository.cs]: I wanted to but i considered that they are two different scenerios thats why I left it

File: CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs (hunk 3 above)
Comment (IruomaJennifer) [reply CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs]: Okay

File: CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs (hunk 4 above)
Comment (IruomaJennifer) [reply CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs]: okay

File: CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs (hunk 5 above)
Comment (IruomaJennifer) [reply CCSA_ChatApplication/Controllers/MessageAndHistoryController.cs]: okay

**Review by Adexandria — APPROVED** [review]

**Pull request merged**